tests/integration/telemetry/stackdriver/vm/vm_test.go

```diff
@@ -45,222 +42,150 @@ func TestVMTelemetry(t *testing.T) {
-				if !srvReceived || !cltReceived {
-					return fmt.Errorf("stackdriver server does not received expected server or client request count, server %v client %v", srvReceived, cltReceived)
-				}
-				if !logReceived {
-					return fmt.Errorf("stackdriver server does not received expected log entry")
-				}
-				if !foundEdge {
-					return fmt.Errorf("stackdriver server did not received expected traffic assertion")
-				}
 				return nil
 			}, retry.Delay(3*time.Second), retry.Timeout(40*time.Second))
 		})
 }
 
-func getWantRequestCountTS() (cltRequestCount, srvRequestCount monitoring.TimeSeries, err error) {
-	srvRequestCountTmpl, err := ioutil.ReadFile(serverRequestCount)
-	if err != nil {
-		return
+func traceEqual(got, want *cloudtrace.Trace) bool {
+	if len(got.Spans) != len(want.Spans) {
+		log.Infof("incorrect number of spans: got %d, want: %d", len(got.Spans), len(want.Spans))
+		return false
 	}
-	sr, err := tmpl.Evaluate(string(srvRequestCountTmpl), map[string]interface{}{
-		"EchoNamespace": ns.Name(),
-	})
-	if err != nil {
-		return
+	if got.ProjectId != want.ProjectId {
+		log.Errorf("mismatched project ids: got %q, want %q", got.ProjectId, want.ProjectId)
+		return false
 	}
-	if err = jsonpb.UnmarshalString(sr, &srvRequestCount); err != nil {
-		return
+
+	for _, wantSpan := range want.Spans {
+		foundSpan := false
+		for _, gotSpan := range got.Spans {
+			delete(gotSpan.Labels, "guid:x-request-id")
+			delete(gotSpan.Labels, "node_id")
+			delete(gotSpan.Labels, "peer.address")
+			delete(gotSpan.Labels, "zone")
+			delete(gotSpan.Labels, "g.co/agent")    // ignore OpenCensus lib versions
+			delete(gotSpan.Labels, "response_size") // this could be slightly off, just ignore
+			if foundSpan = reflect.DeepEqual(gotSpan.Labels, wantSpan.Labels); foundSpan {
+				break
+			}
+		}
+		if !foundSpan {
+			log.Errorf("missing span from trace: got %v\nwant %v", got, want)
+			return false
+		}
 	}
-	cltRequestCountTmpl, err := ioutil.ReadFile(clientRequestCount)
+
+	return true
+}
+
+func gotRequestCountMetrics(wantClient, wantServer *monitoring.TimeSeries) bool {
+	ts, err := sdInst.ListTimeSeries()
 	if err != nil {
-		return
+		log.Errorf("could not get list of time-series from stackdriver: %v", err)
+		return false
 	}
-	cr, err := tmpl.Evaluate(string(cltRequestCountTmpl), map[string]interface{}{
-		"EchoNamespace": ns.Name(),
-	})
-	if err != nil {
-		return
+
+	var gotServer, gotClient bool
+	for _, series := range ts {
+		// Making resource nil, as test can run on various platforms.
+		series.Resource = nil
+		if proto.Equal(series, wantServer) {
+			gotServer = true
+		}
+		if proto.Equal(series, wantClient) {
+			gotClient = true
+		}
 	}
-	err = jsonpb.UnmarshalString(cr, &cltRequestCount)
-	return
+
+	return gotServer && gotClient
 }
 
-func getWantServerLogEntry() (srvLogEntry loggingpb.LogEntry, err error) {
-	srvlogEntryTmpl, err := ioutil.ReadFile(serverLogEntry)
+func gotLogEntry(want *loggingpb.LogEntry) bool {
```

Comment:
for later.. we should combine these with non vm tests too?

Reply:
Yes. probably should try to harmonize both sets of tests.